### DISCUSSION TITLE
Hooks for Apps

### DISCUSSION THREAD

**Discussion by ruscoder** [Aidbox/Issues#475] [General]
Hello! Are there any plans to support hooks for Apps? I mean something like `before-save` that can mutate an original resource through the App.

**Comment by ruscoder** [comment]
I believe it's a very useful feature, e.g. adding some restrictions before resource gets saved or mutate the resource before it gets saved

**Reply by KGOH** [reply]
Pre hooks can be performance killers since they delay operation processing

Could you describe your use-cases in more details, please? Maybe we can solve this in a different way
